package/MDAnalysis/topology/MMTFParser.py

```diff
@@ -0,0 +1,191 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 
+#
+# MDAnalysis --- http://www.MDAnalysis.org
+# Copyright (c) 2006-2015 Naveen Michaud-Agrawal, Elizabeth J. Denning, Oliver Beckstein
+# and contributors (see AUTHORS for the full list)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+MMTF Topology Parser
+====================
+
+Reads topology data from the `Macromolecular Transmission Format
+(MMTF) format`_.  This should generally be a quicker alternative to PDB.
+
+.. versionadded:: 0.16.0
+
+Reads the following topology attributes:
+
+Atoms:
+ - altLoc
+ - atom ID
+ - bfactor
+ - bonds
+ - charge
+ - masses (guessed)
+ - name
+ - occupancy
+ - type
+
+Residues:
+ - icode
+ - resname
+ - resid
+ - resnum
+
+Segments:
+ - segid
+ - model
+
+Classes
+-------
+
+.. autoclass:: MMTFParser
+   :members:
+
+.. _Macromolecular Transmission Format (MMTF) format: https://mmtf.rcsb.org/
+"""
+from six.moves import zip
+
+import mmtf
+import numpy as np
+
+
+from . import base
+from . import guessers
+from ..core.topology import Topology
+from ..core.topologyattrs import (
+    AltLocs,
+    Atomids,
+    Atomnames,
+    Atomtypes,
+    Bfactors,
+    Bonds,
+    Charges,
+    ICodes,
+    Masses,
+    Occupancies,
+    Resids,
+    Resnames,
+    Resnums,
+    Segids,
+    SegmentAttr,  # for model
+)
+from ..core.selection import RangeSelection
+
+
+def _parse_mmtf(fn):
+    if fn.endswith('gz'):
+        return mmtf.parse_gzip(fn)
```

Comment:
Hi, we serve gzipped mmtf files from mmtf.rcsb.org but there is no general purpose compression mandatory for mmtf files. We can discuss adding support for bzip2 to the python library if you require it.